torchao/prototype/parq/optim/quantopt.py

```diff
@@ -106,6 +109,27 @@ def quantize_(
         quants.copy_(Q)
         return q
 
+    @torch.no_grad()
+    def torchao_quantize_(self, model: torch.nn.Module, config: AOBaseConfig):
+        """Recursively call torchao.quantize_ on model using given config."""
+        self.restore_latent_params()
+        param_set = {
+            p.data_ptr()
+            for group in self.regularized_param_groups()
+            for p in group["params"]
+        }
+
+        def inner_quantize_(model):
+            for module in model.children():
+                for param in module.parameters(recurse=False):
+                    if param.data_ptr() in param_set:
+                        quantize_(module, config)
```

Comment:
By the way this call itself is also recursive, so I feel this will end up quantizing the same submodules multiple times? Maybe we should express this as part of a filter_fn, something like:

```
def _filter_fn(module: torch.nn.Module) -> bool:
    for param in module.parameters(recurse=False):
        if param.data_ptr() in param_set:
            return True
    return False

quantize_(model, config, _filter_fn)
```

Reply:
Good point! I'll add this change in